sqlglot/parser.py

```diff
@@ -5475,6 +5476,53 @@ def _parse_comprehension(
             condition=condition,
         )
 
+    def _parse_heredoc(self) -> t.Optional[exp.Heredoc]:
+        tags = []
+
+        if self._match_text_seq("$"):
+            tags.append("$")
+        else:
+            return None
+
+        if self._curr and self._is_connected():
+            if not self._match_text_seq("$"):
+                tag_text = self._curr.text
+                tags.append(tag_text)
+                self._advance()
+            else:
+                tag_text = None
+                tags.append("$")
+        else:
+            self.raise_error("No closing $ found")
+
+        if tag_text:
+            if self._is_connected() and self._match_text_seq("$"):
+                tags.append("$")
+            else:
+                self.raise_error("No closing $ found")
+
+        heredoc_start = self._curr
+
+        while self._curr:
+            index = self._index
+            heredoc_end = self._prev
+            tags_matched = True
+
+            for tag in tags:
```

Comment:
why don't you do match_text_seq here?

Reply:
`match_text_seq` would convert everything to upper case, meanwhile we want the beginning and closing tags to be the same, case sensitive (e.g. `$foo$` and `$FOO$` shouldn't match)

Reply:
as i said, i don't think this edge case matters much, so let's just use match_text_seq